ethpm/package.py

```diff
@@ -130,7 +132,7 @@ def manifest_version(self) -> str:
         return self.manifest["manifest_version"]
 
     @property
-    def uri(self) -> str:
+    def uri(self) -> Optional[str]:
```

Comment:
Why `Optional`?  Assuming this is to take into account packages initialized without a `URI`, what if we just generated a [data URI](https://en.wikipedia.org/wiki/Data_URI_scheme) for those?  Data URIs could **always** be valid since they contain their content.